src/ui_iteminfo.cpp

```diff
@@ -0,0 +1,195 @@
+#include "ui_iteminfo.h"
+
+#include "messages.h"
+#include "ui_manager.h"
+#include "imgui/imgui_internal.h"
+
+
+static void set_scroll( scroll &s )
+{
+    int scroll_px = 0;
+    int line_height = ImGui::GetTextLineHeightWithSpacing();
+    int page_height = ImGui::GetCurrentContext()->CurrentWindow->InnerRect.GetHeight();
```

Comment:
Oh, probably because `set_scroll` is called after putting everything into the window; the negative value is literally how far past the bottom of the window you are.

Reply:
Right, it gives proper values if you set the scroll before drawing. A difference of 16px remains between the two versions. For comparison line height with spacing is 20px. So current version scrolls a whole page, your version basically keeps the last line in view when scrolling down. I don't really have a preference between the two, but I guess yours is the more common behavior?

Reply:
That difference is probably only because `GetContentRegionAvail()` subtracts the padding inside the window, whereas `InnerRect` includes it. I think you would get the same answer if you compared it with `WorkRect` instead. But I have only a passing familiarity with these internals.

Leaving the bottom line in view is a great idea because it lets the user retain their context better. However, it would be better to do it on purpose, rather than accidentally :D

Reply:
WorkRect seems to be something completely different, that's giving me a very large number, possibly the size of all the contents.

Doing it on purpose would mean content aware scrolling, which is a little hard since separators and spacing are smaller than text. Though there's the simple option of subtracting line_height from page_height (using InnerRect).

But we're kinda getting lost in details here. I'll just switch it over to `GetContentRegionAvail`.

Reply:
No, even I wouldn't go to that level of detail. The key is just to scroll a consistent amount but leave something from the bottom of the window visible at the top after the scroll. The exact amount varies quite widely across many programs. Some aim to leave about one line of text, others some fraction of the page, others none. Emacs only scrolls by half of the visible window, for example, on the theory that more context is more better.